<launch>
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
      <remap from="cloud_in" to="/point_cloud"/>
      <remap from="scan" to="/scan_msgs" />
      <rosparam>
        target_frame: /sensor_base_link
        transform_tolerance: 0.01
        min_height: 0.0
        max_height: 1.0
        
        angle_min: -0.28686334952 # atan( (175 - 57) / 400
        angle_max: 0.28686334952 
        angle_increment: 0.0007968 # angle_max / 180
        scan_time: 0.01
        range_min: -100
        range_max: 500
        use_inf: true
        
        #concurrency_level affects number of pc queued for processing and the number of threadsused
        # 0: Detect number of cores
        # 1: Single threaded
        # 2: inf : Parallelism level
        concurrency_level: 1
      </rosparam>
    </node>
</launch>
